tests/system_tests_tcp_conns_terminate.py

```diff
@@ -0,0 +1,387 @@
+                    return rec['IDENTITY']
+        except KeyError:
+            pass
+        return None
+
+    def run_skmanage(self, cmd, router):
+        p = self.popen(
+            ['skmanage'] + cmd.split(' ') + ['--bus', router.addresses[0],
+                                             '--indent=-1', '--timeout', str(TIMEOUT)],
+            stdin=PIPE, stdout=PIPE, stderr=STDOUT, universal_newlines=True)
+        out = p.communicate()[0]
+        try:
+            p.teardown()
+        except Exception as e:
+            raise Exception(out if out else str(e))
+        return out
+
+    def delete_tcp_entity(self, address, entity_type, router):
+        def find_entity(address, entity_type, router, expected=True):
+            query_cmd = f'QUERY --type={entity_type}'
+            output = json.loads(self.run_skmanage(query_cmd, router))
+            res = [e for e in output if e['address'] == address]
+            if expected:
+                self.assertTrue(len(res) == 1)
+                return res[0]
+            else:
+                self.assertTrue(len(res) == 0)
+            return None
+
+        e = find_entity(address, entity_type, router)
+        name = e['name']
+        delete_cmd = 'DELETE --type=' + entity_type + ' --name=' + name
+        self.run_skmanage(delete_cmd, router)
+        # check that rthe entity has been deleted
+        retry_assertion(lambda: find_entity(address, entity_type, router, expected=False),
+                        timeout=2, delay=1)
+
+    def create_echo_clients(self, client_prefix, client_port):
+        # We use the delay_close flag to keep the connections open
+        echo_clients = []
+        client_logger = Logger(title=client_prefix,
+                               print_to_console=True)
+        for i in [1, 2]:
+            echo_clients.append(TcpEchoClient(client_prefix + '_' + str(i),
+                                              host='localhost',
+                                              port=client_port,
+                                              size=1,
+                                              count=1,
+                                              logger=client_logger,
+                                              delay_close=True))
+        return echo_clients
+
+    def clean_up_echo_clients(self, echo_clients):
+        for e in echo_clients:
+            e.wait()
+
+    def setup_flows(self, address):
+        """
+        Setup tcp flows via the tcpListeners and the tcpConnector which have
+        the particular VAN address. Two flows are created for each tcpListener.
+        """
+        router_1_id = self.router_1.config.router_id
+        router_2_id = self.router_2.config.router_id
+
+        # vflow Ids are used to associate flows with tcpListeners and tcpConnectors
+        vflow_ids = {}
+        self.assertTrue(self.router_1_vflow_id is not None)
```

Comment:
You can use `self.assertIsNotNone` instead of `self.assertTrue`

Reply:
Done